lib/split/helper.rb

```diff
@@ -44,19 +45,20 @@ def reset!(experiment)
 
     def finish_experiment(experiment, options = {:reset => true})
       return true if experiment.has_winner?
+      experiment.version = Split::Experiment.version_for_experiment_key(ab_user.key_for_experiment(experiment)).to_i
       should_reset = experiment.resettable? && options[:reset]
-      if ab_user[experiment.finished_key] && !should_reset
+      if ab_user[experiment.finished_key(ab_user.version_for_experiment(experiment))] && !should_reset
         return true
       else
-        alternative_name = ab_user[experiment.key]
+        alternative_name = ab_user[Split::Experiment.key_without_version_for_experiment_key(experiment.key)]
         trial = Trial.new(:user => ab_user, :experiment => experiment,
-              :alternative => alternative_name)
+              :alternative => alternative_name, :version => Split::Experiment.version_for_experiment_key(ab_user.key_for_experiment(experiment)))
         trial.complete!(options[:goals], self)
 
         if should_reset
           reset!(experiment)
         else
-          ab_user[experiment.finished_key] = true
+          ab_user[experiment.finished_key(ab_user.version_for_experiment(experiment))] = true
```

Comment:
Is passing in the result of `experiment.finished_key(ab_user.version_for_experiment(experiment))` to the `finished_key` function necessary? Would't the experiment object have the appropriate version from it being set in line 48?